client.go

```diff
@@ -189,12 +189,17 @@ func (d BaseClient) doRequest(httpRequest *http.Request, target any, opts ...Opt
 		if d.config.Non2xxStatusHandler != nil {
 			d.config.Non2xxStatusHandler(httpResponse, data)
 		}
+		if err = checkForOperationOutcome(data); err != nil {
+			return err
+		}
 		return fmt.Errorf("FHIR request failed (%s %s, status=%d)", httpRequest.Method, httpRequest.URL.String(), httpResponse.StatusCode)
 	}
 	if len(data) > d.config.MaxResponseSize {
 		return fmt.Errorf("FHIR response exceeds max. safety limit of %d bytes (%s %s, status=%d)", d.config.MaxResponseSize, httpRequest.Method, httpRequest.URL.String(), httpResponse.StatusCode)
 	}
-	// TODO: Handle errornous responses (OperationOutcome?)
+	if err = checkForOperationOutcome(data); err != nil {
```

Comment:
> Operation outcomes returned SHOULD be in alignment with the HTTP response code. For example, if the HTTP code indicates a failure (300+), at least one of the issues should have a severity of "error", indicating the reason for the failure.
(https://www.hl7.org/fhir/r4b/operationoutcome.html#using)

and

> Operation outcomes are sets of error, warning and information messages that provide detailed information about the outcome of an attempted system operation. They are provided as a direct system response, or component of one, and provide information about the outcome of the operation.

(note "informational")

In other words, an operation outcome for an OK status could be considered as the opertion succeeded? I think the response the FHIR server returns when creating a Bundle (FHIR transaction) also contains an OperationOutcome.